src/main/java/org/opensearch/flowframework/workflow/AbstractRetryableWorkflowStep.java

```diff
@@ -110,25 +91,36 @@ void retryableGetMlTask(
                 try {
                     logger.info(workflowStep + " successful for {} and modelId {}", workflowId, response.getModelId());
                     String resourceName = WorkflowResources.getResourceByWorkflowStep(getName());
+                    String id;
                     if (getName().equals(WorkflowResources.DEPLOY_MODEL.getWorkflowStep())) {
-                        completeFuture(resourceName, nodeId, workflowId, response, future);
+                        id = response.getModelId();
```

Comment:
By the time we get here, MLTaskState is COMPLETED in both register and deploy model.  So both have a model ID.  Why are we treating the task Id differently for them?  Shouldn't it be `modelId` for both of them?

Interestingly the existing code uses `response.getTaskId()` for the resource ID, but the output is clearly the model Id.

Is taskid = modelid for an initial model registration?

Reply:
> Is taskid = modelid for an initial model registration?

It is not the same, a completed task will give us a model ID, the task ID is merely the document ID of the ML Task index.

Reply:
will change to both use modelId in follow up PR

Reply:
Talked with @amitgalitz offline, we've combined the `RegisterLocalModelStep` and `GetMLTaskStep` to make the call effectively synchronous within our framework. The task ID is returned by the local model registration API call, and we use this to ascertain the status of the model deployment, which gives us the model ID once it is in a completed state. The output of this RegisterLocalModelStep should just be the `model_id`, the task id is just used internally within the step